plugin/main/src/kotlinx/benchmark/gradle/JmhBytecodeGeneratorTask.kt

```diff
@@ -8,10 +9,14 @@ import org.gradle.workers.*
 import java.io.*
 import javax.inject.*
 
-@Suppress("UnstableApiUsage")
 @CacheableTask
 open class JmhBytecodeGeneratorTask
```

Comment:
I'd recommend not doing this. The tasks are part of the public API, and are accessible to users. We want to makes sure that any changes we made to tasks would be monitored.

When a user would like to manually configure a task, it would require opt-ins, which would be annoying.

It would also trigger warnings for generated Kotlin DSL accessors.

Reply:
> The tasks are part of the public API, and are accessible to users.

I am not sure if it is a good idea to keep it public and accessible to users. The preferred way of configuring benchmarks is through `benchmark` extension, and most of the task properties are derive from settings inside `benchmark` extension.

Also, I searched the grep.app and didn't find direct configuration of the plugin tasks.

Another benefit of hiding the tasks is later we can modernise its API without migration path. And once the API is baked, we could make it public again.

> It would also trigger warnings for generated Kotlin DSL accessors.

Is it possible to configure how the accessors are generated? It would be nice if such tasks had `DefaultTask` type in public.

Reply:
I think those are all relevant points that should be addressed, but can we look at them later? Can you make an issue on GitHub/YouTrack to keep track of the proposal?

In this PR, I'd like to just focus on adding the API dump, and make some minor changes to 'soft deprecate' some internal functionality. Currently, the tasks are public, and so the API dump should reflect that.

Reply:
Created an issue to track the proposal: https://github.com/Kotlin/kotlinx-benchmark/issues/199